### PR TITLE
Add docs for azure_network_security_group(s) resources

### PR DESCRIPTION
**Pull request by Padgett**
Signed-off-by: Joshua Padgett <jpadgett@chef.io>

### CODE REVIEW

File: docs/resources/azure_network_security_group.md.erb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Examples
+
+### Test that an example Resource Group has the specified Network Security Group
+
+    describe azure_network_security_group(resource_group: 'example', name: 'GroupName') do
+      it { should exist }
+    end
+
+### Test that an example Resource Group has a Network Security Group that allows ssh from the Internet
```
Comment (TrevorBramble) [inline docs/resources/azure_network_security_group.md.erb]: "SSH" and "internet" (sadly)

File: docs/resources/azure_network_security_groups.md.erb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Syntax
+
+An `azure_network_security_groups` resource block identifies Network Security
+Groups by Resource Group.
+
+    describe azure_network_security_groups(resource_group: 'ExampleGroup') do
+      ...
+    end
+<br />
```
Comment (TrevorBramble) [inline docs/resources/azure_network_security_groups.md.erb]: Still don't know why we need these awkward HTML line break elements, but the pattern is to put an empty line above and below.

File: docs/resources/azure_network_security_group.md.erb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    end
+
+## Properties
+
+* `security_rules`
+* `default_security_rules`
+* `allow_ssh_from_internet`
+* `allow_rdp_from_internet`
+
+### security_rules
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+The security_rules property contains the set of Security Rules.
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    its('security_rules') { should_not be_empty }
+
+### default_security_rules
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+The default_security_rules property contains the set of Default Security Rules.
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    its('default_security_rules') { should_not be_empty }
+
+### allow_ssh_from_internet
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+The allow_ssh_from_internet property contains a boolean value determined by analysing
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+the Security Rules and Default Security Rules for unrestricted SSH access.
+
+    it { should_not allow_ssh_from_internet }
+
+### allow_rdp_from_internet
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

File: docs/resources/azure_network_security_group.md.erb [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+The allow_rdp_from_internet property contains a boolean value determined by analysing
```
Comment (dmccown) [inline docs/resources/azure_network_security_group.md.erb]: I recommend escaping the underscore (`\_`).

**Pull request merged**